test/integration/puppetlabs/general_puppet/general_puppet_int_test.clj

```diff
@@ -273,4 +279,18 @@
      app {:jruby-puppet
           {:max-active-instances num-jrubies}}
      (let [catalog (testutils/get-catalog)]
-       (is (= "production" (get catalog "version")))))))
\ No newline at end of file
+       (is (= "production" (get catalog "version")))))))
+
+(deftest ^:integration custom-trusted-oid-mapping-test
+  (testing "custom_trusted_oid_mapping works properly"
+    (testutils/write-site-pp-file
+      "notify { 'trusted_hash':\n\tmessage => \"${trusted['extensions']}\",\n}")
+    (fs/copy custom-oid-mapping (fs/file testutils/conf-dir "custom_trusted_oid_mapping.yaml"))
```

Comment:
I think it would be good to ensure that these files get cleaned up after the test is done running.

One option might be to just add some variables in the `puppet.conf` in the `general_puppet_int_test` dev-resources directory which point at the specific locations of the `csr-attributes` and `custom-trusted-oid-mapping` files.  Then when the `puppet.conf` file is cleaned out by the text fixture already in place for this file, any tests run outside of this namespace wouldn't use these extra files.